addons/hr_holidays/models/hr_leave.py

```diff
@@ -788,7 +788,7 @@ def _check_validity(self):
                 max_excess = leave_type.max_allowed_negative
                 for employee in employees:
                     if leave_data[employee] and leave_data[employee][0][1]['virtual_remaining_leaves'] < -max_excess:
-                        raise ValidationError(_("There is no valid allocation to cover that request."))
+                        raise ValidationError(_("The employee %s does not have a valid allocation for the leave type %s to cover that request.", employee.name, leave_type.name))
```

Comment:
I think the name of the employee is sufficient :)
Additionally, you need to name the text injections so that it is independent of the order of those injections in the sentence (since it might vary from a language to another)
```suggestion
                        raise ValidationError(_("%(name)s does not have a valid allocation for the leave type %(leave_type)s to cover that request.", name=employee.name, leave_type=leave_type.name))
```